### ISSUE TITLE
Sherpa does not accurately extract numeric values from search results

### ISSUE THREAD

**Issue by YujingYang666777** [Aggregate-Intellect/sherpa#80]
Description: when searching for the annual salary for several jobs, the results given by the LLM are exactly the same. Probably the searched result can be more accurate regarding the specific question.

For example: 
User: @Sherpa What is the annual salary for an entry level software engineer in Canada?
Sherpa: 
The average annual salary for an entry-level software engineer in Canada ranges from $68,207 to $110,963. Source: [Levels.fyi](https://www.levels.fyi/t/software-engineer/levels/entry-level/locations/canada)

However, when clicking into the URL, the number display on the website is 5.07 - 11.13 instead of the number claimed by Sherpa.

**Comment by amirfz** [comment]
@YujingYang666777 could you add some examples to the description?

**Comment by amirfz** [comment]
@YujingYang666777 talk to @johnnykfeng because he had to solve this issue in a different project

**Comment by oshoma** [comment]
@YujingYang666777 @johnnykfeng is this issue still occurring? Any comments to add?

**Comment by oshoma** [comment]
I just tried this again. It's still happening.
```
Osh: https://github.com/sherpa What is the annual salary for an entry level software engineer in Canada?
Sherpa: The annual salary for an entry level software engineer in Canada can vary. According to Glassdoor, the median pay for entry level software engineers in Canada is CA$104,203 [[1]](https://www.levels.fyi/t/software-engineer/levels/entry-level/locations/canada). Another source, Levels.fyi, states that the average salary range for entry level software engineers in Canada is from CA$67,576 to CA$107,680 [[2]](https://www.glassdoor.ca/Salaries/canada-entry-level-software-engineer-salary-SRCH_IL.0,6_IN3_KO7,36.htm). It's important to note that salaries can also depend on factors such as location and experience.
```

The $104,203 number does not appear anywhere on the 1st URL page.
For the 2nd URL, the actual range on the web page is $84K - $130K/yr.

**Comment by amirfz** [comment]
part of the evidence verification that Suhas has sent me is dealing with
numbers and entities specifically. maybe once we have a baseline "evidence
verification" and "citation" then we can break this down more specifically
to sub tasks that need to be handled?

**Comment by amirfz** [comment]
ok, to add more concrete details. this is also related and can probably borrow implementation ideas from #241  
- create a critic agent for the QA agent @20001LastOrder 
- extract numbers from the response (regex?) @Eyobyb 
- check if those numbers exist exactly as they are in the references then accept the answer and send it to the user @Eyobyb 
- if not, then respond back to the QA agent with something like "the numbers presented in your answer do not appear in the references. please try again responding and including the correct numbers"

ps. this can be also done for entities that are important.
pps. this doesn't solve the problem of a number existing but not being what the user actually wanted, but this would be a good baseline for obvious errors

**Comment by oshoma** [comment]
@Eyobyb is working on #260 in relation to this.

**Comment by 20001LastOrder** [comment]
Closed by #277
